File: Paradigmas/Atv-09_06_folha_pagamento/python/main.py
import folha_pagamento as folha

QUANTIDADE_PROFESSORES = 2
MAX_HORAS = 60


def main():
    lista_professores = []

    print("SISTEMA DE FOLHA DE PAGAMENTO\n")

    for i in range(QUANTIDADE_PROFESSORES):
        print(f"Professor {i + 1}")
        nome = input("Nome: ")
        horas = float(input("Horas semanais: "))
        while horas > MAX_HORAS:
            print(f"Horas invalidas! O maximo e {MAX_HORAS}.")
            horas = float(input("Horas semanais: "))
        valor_emprestimo = float(input("Valor do emprestimo (0 se nao tiver): "))
        print()

        professor = folha.processar_professor(nome, horas, valor_emprestimo)
        lista_professores.append(professor)

    print("\nCONTRACHEQUES:\n")
    for professor in lista_professores:
        folha.imprimir_contracheque(professor)

    folha.gerar_relatorio_geral(lista_professores)


main()

File: Paradigmas/Atv-09_06_folha_pagamento/javascript/main.js
const folha = require("./folha_pagamento");
const readline = require("node:readline/promises");
const { stdin: input, stdout: output } = require("node:process");

const QUANTIDADE_PROFESSORES = 2;
const MAX_HORAS = 60;


async function main() {
    const rl = readline.createInterface({ input, output });
    const listaProfessores = [];

    console.log("SISTEMA DE FOLHA DE PAGAMENTO\n");

    for (let i = 0; i < QUANTIDADE_PROFESSORES; i++) {
        console.log(`Professor ${i + 1}`);
        const nome = await rl.question("Nome: ");
        let horas = parseFloat(await rl.question("Horas semanais: "));
        while (horas > MAX_HORAS) {
            console.log(`Horas invalidas! O maximo e ${MAX_HORAS}.`);
            horas = parseFloat(await rl.question("Horas semanais: "));
        }
        const valorEmprestimo = parseFloat(await rl.question("Valor do emprestimo (0 se nao tiver): "));
        console.log();

        const professor = folha.processarProfessor(nome, horas, valorEmprestimo);
        listaProfessores.push(professor);
    }

    console.log("\nCONTRACHEQUES:\n");
    for (const professor of listaProfessores) {
        folha.imprimirContracheque(professor);
    }

    folha.gerarRelatorioGeral(listaProfessores);

    rl.close();
}


main();
